### ISSUE TITLE
Feature: Notification panel

### ISSUE THREAD

**Issue by asudoh** [carbon-design-system/carbon-custom-elements#299]
https://www.carbondesignsystem.com/patterns/notification-pattern/#notification-panel

**Comment by xxxle0** [comment]
hi @asudoh, can i contribute this component ?

**Comment by asudoh** [comment]
Definitely, thank you for jumping in @xxxle0! Great if you can read [the coding convention doc](https://github.com/carbon-design-system/carbon-custom-elements/blob/master/src/coding-conventions.md) beforehand. Also there are some coding conventions I haven't been able to document yet, so if you can read some of the existing code and take cues from there it'll be great. Please don't hesitate to ask any time if you have any questions!

**Comment by xxxle0** [comment]
ok. thanks 👍

**Comment by xxxle0** [comment]
hi @asudoh, i read the codebase and this repo will inherit css from carbon package component. Just convert the presentational component to Web component. Is it right ?

**Comment by asudoh** [comment]
Thanks you for asking @xxxle0 - Let's take `<bx-dropdown-item>` as an example. It has [static `styles` property](https://github.com/carbon-design-system/carbon-custom-elements/blob/v1.0.0-beta.3/src/components/dropdown/dropdown-item.ts#L57) that dictates the style set to shadow DOM, [coming from `dropdown.scss` from the same directory](https://github.com/carbon-design-system/carbon-custom-elements/blob/v1.0.0-beta.3/src/components/dropdown/dropdown-item.ts#L12). The [`dropdown.scss`](https://github.com/carbon-design-system/carbon-custom-elements/blob/v1.0.0-beta.3/src/components/dropdown/dropdown.scss) file [imports several styles from Carbon](https://github.com/carbon-design-system/carbon-custom-elements/blob/v1.0.0-beta.3/src/components/dropdown/dropdown.scss#L10-L13) as you may have alluded to, but it also has [styles for `:host(#{$prefix}-dropdown-item)` selector](https://github.com/carbon-design-system/carbon-custom-elements/blob/v1.0.0-beta.3/src/components/dropdown/dropdown.scss#L38-L59). This is because `<bx-dropdown-item>` is an actual DOM element unlike React component node (e.g. `carbon-components-react`'s [`<ListBoxMenuItem>`](https://github.com/carbon-design-system/carbon/blob/v10.9.0/packages/react/src/components/ListBox/ListBoxMenuItem.js)) which is _not_ a DOM element. That means the style of `<bx-dropdown-item>` has to be mapped to the top-level element `<ListBoxItem>` renders, which is, [`<div className="bx--list-box__menu-item">`](https://github.com/carbon-design-system/carbon/blob/v10.9.0/packages/react/src/components/ListBox/ListBoxMenuItem.js#L27). Fortunately many cases for `<bx-dropdown-item>` it's as simple as using Sass `@extend` along with fixing `display` where we want `block` but browser's default style for unknown element is `inline`. But as you see in some style rules there (e.g. [one for disabled dropdown item](https://github.com/carbon-design-system/carbon-custom-elements/blob/v1.0.0-beta.3/src/components/dropdown/dropdown.scss#L43-L46)) we may need more manual styling depending on how the corresponding Carbon style is defined.

Other than that, it's mostly about having `lit-element` define properties, render the contents, and handling events. Don't hesitate to ask any other questions. Thanks!

**Comment by oliviaflory** [comment]
I have definitely seen a request for this or something similar to this before from someone attending Design Office Hours.
